zipline/pipeline/factors/statistical.py

```diff
@@ -66,32 +73,38 @@ class RollingPearson(_RollingCorrelation):
     Most users should call Factor.pearsonr rather than directly construct an
     instance of this class.
     """
-    def compute(self, today, assets, out, factor_data, slice_data):
-        slice_data_column = slice_data[:, 0]
+    def compute(self, today, assets, out, base_data, target_data):
+        # If `target_data` is a Slice or single column of data, broadcast it
+        # out to the same shape as `base_data`, then compute column-wise. This
+        # is efficient because each column of the broadcasted array only refers
+        # to a single memory location.
+        target_data = broadcast_arrays(target_data, base_data)[0]
```

Comment:
@ssanderson I'm using `broadcast_arrays` here because `broadcast_to` was not introduced until numpy version 1.10.0